From 8d0e46c1ddb5b7f0992591b0dc5d8aaa77cc9dba Mon Sep 17 00:00:00 2001
From: Alkaid <zgf574564920@gmail.com>
Date: Tue, 12 Mar 2024 03:59:12 -0700
Subject: [PATCH] Fix OpenSSL ED25519 support detection

Wrong function signature in configure.ac prevents openssh from enabling
the recently new support for ED25519 priv keys in PEM PKCS8 format.
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 82e8bb7c141..081e2bc75bd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3184,7 +3184,7 @@ if test "x$openssl" = "xyes" ; then
 		]], [[
 		unsigned char buf[64];
 		memset(buf, 0, sizeof(buf));
-		exit(EVP_PKEY_new_raw_private_key(EVP_PKEY_ED25519,
+		exit(EVP_PKEY_new_raw_private_key(EVP_PKEY_ED25519, NULL,
 		    buf, sizeof(buf)) == NULL);
 		]])],
 		[
